python_modules/dagster-webserver/dagster_webserver_tests/webserver/test_app.py

```diff
@@ -278,3 +287,163 @@ def test_download_compute(instance, test_client: TestClient):
+    response = test_client.post(
+        f"/report_asset_materialization/{my_asset_key}",
+        params={
+            # "description": mat.description,
+            "partition": mat.partition,
+            "metadata": json.dumps(meta),
+            "data_version": "new",
+        },
+    )
+    assert response.status_code == 200, response.json()
+    evt = instance.get_latest_materialization_event(AssetKey(my_asset_key))
+    assert evt
+    assert evt.asset_materialization
+    assert evt.asset_materialization == mat
+
+    # bad metadata
+    response = test_client.post(
+        f"/report_asset_materialization/{my_asset_key}",
+        params={
+            "metadata": meta,  # not json encoded
+        },  # type: ignore
+    )
+    assert response.status_code == 400
+    assert "Error parsing metadata json" in response.json()["error"]
+
+    response = test_client.post(
+        f"/report_asset_materialization/{my_asset_key}",
+        json={
+            "metadata": "im_just_a_string",
+        },
+    )
+    assert response.status_code == 400
+    assert (
+        'Error constructing AssetMaterialization: Param "metadata" is not'
+        in response.json()["error"]
+    )
+
+
+def test_report_asset_materialization_apis_consistent(
+    instance: DagsterInstance, test_client: TestClient
+):
+    # ensure the ext report_asset_materialization and the API endpoint have the same capabilities
+    sample_payload = {
+        "asset_key": "sample_key",
+        "metadata": {"meta": "data"},
+        "data_version": "so_new",
+        "partition": "2023-09-23",
+    }
+
+    # sample has entry for all supported params (banking on usage of enum)
+    assert set(sample_payload.keys()) == set({e.value for e in ReportAssetMatParam})
+
+    # all ext report_asset_materialization kwargs should be in sample payload
+    sig = inspect.signature(ExtContext.report_asset_materialization)
+    skip_set = {
+        "self",
+    }
+    params = [p for p in sig.parameters if p not in skip_set]
+    assert set(sample_payload.keys()) == set(params)
+
+    response = test_client.post("/report_asset_materialization/", json=sample_payload)
+    assert response.status_code == 200
+    evt = instance.get_latest_materialization_event(AssetKey(sample_payload["asset_key"]))
+    assert evt
+    mat = evt.asset_materialization
+    assert mat
+
+    for k, v in sample_payload.items():
+        if k == "asset_key":
+            assert mat.asset_key == AssetKey(v)
+        elif k == "metadata":
+            assert mat.metadata.keys() == v.keys()
+        elif k == "data_version":
+            tags = mat.tags
+            assert tags
+            assert tags[DATA_VERSION_TAG] == v
+            assert tags[DATA_VERSION_IS_USER_PROVIDED_TAG]
+        elif k == "partition":
+            assert mat.partition == v
+        else:
+            assert (
+                False
+            ), "need to add validation that sample payload content was written successfully"
```

Comment:
I setup this test to keep things in sync but `ExtContext.report_asset_materialization`  does not support `partition` I assume since its banking on just filling that in based on the context of what happening in the run, but we can't do that over here.

Should ext support `partition` or should i just mark it as ok to be out of sync ?